include/ceed-impl.h

```diff
@@ -28,7 +28,7 @@
 #define CEED_MAX_RESOURCE_LEN 1024
 #define CEED_ALIGN 64
 
-#define CEED_NUM_BACKEND_FUNCTIONS 29
+#define CEED_NUM_BACKEND_FUNCTIONS 30
```

Comment:
Future: We should use `offsetof` to avoid magic numbers that need to be maintained.

Reply:
I'm using `offsetof` to calculate the lookup table of backend functions; this number is the total number of backend functions across all objects. I guess I'm not sure how `offsetof` helps me know how many function pointers are in all of the object?

Reply:
1. Why `ceedoffsetof` when `offsetof` is defined in `stddef.h`?
2. The translation table does not need to be allocated statically.

Reply:
1. I was having difficulty getting `offsetof` and our opaque pointers to work together.
2. Good point - I can fix that up pretty easily.

Reply:
Syntax would be `offsetof(struct Ceed_private, Error)`. Note that you can avoid repetition by using a macro with stringify `#` and concatenation `##`.

Reply:
Also, `CeedSetBackendFunction` should not mix data and function pointers. Better something like this:

```c
@@ -482,9 +482,8 @@ int CeedSetBackendFunction(Ceed ceed,
   for (CeedInt i = 0; i < CEED_NUM_BACKEND_FUNCTIONS; i++) {
     if (!strcmp(ceed->foffsets[i].fname, lookupname)) {
       size_t offset = ceed->foffsets[i].offset;
-      size_t *fpointer;
-      fpointer = (size_t *)(object + offset);
-      *fpointer = (size_t) f;
+      int (**fpointer)(void) = (int (**)(void))((char*)object + offset);
+      *fpointer = f;
       return 0;
     }
   }
```

Reply:
I was thinking something like (untested)
```c
#define CEED_FTABLE_ENTRY(class, method) \
  {#class #method, offsetof(struct class ## _private, method)}
```
then
```c
  foffset foffsets[] = {
    CEED_FTABLE_ENTRY(Ceed, Error),
    ...
    CEED_FTABLE_ENTRY(Ceed, ElemRestrictionCreateBlocked),
    ...
    CEED_FTABLE_ENTRY(CeedVector, SetArray),
```

Reply:
Added in PR #232